postgresql/resource_postgresql_extension.go

```diff
@@ -101,8 +121,16 @@ func resourcePostgreSQLExtensionExists(d *schema.ResourceData, meta interface{})
 	defer c.catalogLock.Unlock()
 
 	var extensionName string
+
+	database := getDatabaseForExtension(d, c)
+	txn, err := startTransaction(c, database)
```

Comment:
You need to check that db exists before trying to connect on it:

```go
       // Check the database exists
       exists, err := dbExists(c.DB(), database)
       if err != nil || !exists {
               return false, err
       }
```
[`dbExists`](https://github.com/terraform-providers/terraform-provider-postgresql/blob/62bb8a15977f56dbebcb5570a41bbe3e5146f0e7/postgresql/helpers.go#L156) can be modified to support both `sql.Tx` and `sql.DB` by creating an interface (already had to do that and inspired from an issue in `database/sql`):

```go
// QueryAble is common interface for sql.DB and sql.Tx
// (cf https://github.com/golang/go/issues/14468)
type QueryAble interface {
	Exec(query string, args ...interface{}) (sql.Result, error)
	Query(query string, args ...interface{}) (*sql.Rows, error)
	QueryRow(query string, args ...interface{}) *sql.Row
}

[...]
func dbExists(db QueryAble, dbname string) (bool, error) {
[...]
```